### ISSUE TITLE
Psx games stutters

### ISSUE THREAD

**Issue by cresi** [OpenEmu/OpenEmu#1868]
I opened a thread on reddit about this issue i'm having emulating psx games but no one could help. When I open psx games they start fine (PSX and Sony logo load just fine), but when i'm actually at menu screen or further into gameplay the game stutters (FPS spikes and sound distortion). PS1 bios are ok as i followed guidelines, I've tried to select all the filters but no one seems to fix the issue and i've tried to run the emulator without any app opened but it does nothing. Two days ago the problem seemed fixed but it came back the day after. 

I have an iMac with these specs: 
-OSX v 10.10.1
-Intel Core i7 3.4 Ghz
-8 GB 1333 Mhz DDR3
-AMD Radeon HD 6970M 2048 MB

Hope you can help me fixing this issue.

**Comment by clobber** [comment]
While you're having the issue, run "sudo spindump" in Terminal and then link the file it generates back here via gist.github.com, dropbox or transfer.sh

**Comment by cresi** [comment]
Here the file:
https://www.dropbox.com/s/mzwocc0jf0dgxnw/spindump.txt?dl=0

**Comment by clobber** [comment]
Mednafen is using most of the CPU time. Which version of the Mednafen core port are you using and did you compile it yourself?

**Comment by cresi** [comment]
I'm using the one that openemu installs when you run a psd game for the first time, the version 0.9.34.1

**Comment by clobber** [comment]
Okay, well the backtrace from the spindump is rather complete and makes it appear the core was built in debug, which would happen if one compiled the core on their own and didn't build for optimization.

Anyway, 0.9.34.1 is old and there could be some performance issues (we don't yet ship the latest Medanfen version until next OpenEmu release). What happens if you try this newer version?

https://github.com/OpenEmu/Mednafen-Core/releases/download/v0.9.38.2/Mednafen_0.9.38.2.zip

Download, unzip and place the .oecoreplugin file in ~/Library/Application Support/OpenEmu/Cores (~ meaning your Home folder where ~ is your username. Guide here if you don't know this: http://osxdaily.com/2014/12/16/show-user-library-folder-os-x-yosemite/)

**Comment by cresi** [comment]
It gives me this error:
Couldn't communicate with a helper application.
Try your operation again. If that fails, quit and relaunch the application and try again.

**Comment by clobber** [comment]
Huh?

Okay, run `find -s ~/Library/Application\ Support/OpenEmu/BIOS -type f -exec md5 '{}' \;` in Terminal and paste the results back here.

I also need to know which game you're trying to load and what the cuesheet and files look like. So in "~/Library/Application Support/OpenEmu/Game Library/roms/Sony PlayStation" copy and paste the names of all the files for the game (.bin, .cue) and also open up the .cue file in TextEdit for the game in question and paste its contents back here.

Finally, open up Console.app and look for the latest crash log that should be there for "OpenEmuHelperApp" and open that up and paste its contents in gist.github.com and link that back here too.

**Comment by cresi** [comment]
find -s ~/Library/Application\ Support/OpenEmu/BIOS -type f -exec md5 '{}' \;
MD5 (/Users/stefanodimarino/Library/Application Support/OpenEmu/BIOS/.DS_Store) = 7ced86b3e6fdea4082e42c7842e8068d
MD5 (/Users/stefanodimarino/Library/Application Support/OpenEmu/BIOS/SCPH5500.BIN) = 8dd7d5296a650fac7319bce665a6a53c
MD5 (/Users/stefanodimarino/Library/Application Support/OpenEmu/BIOS/SCPH5501.BIN) = 490f666e1afb15b7362b406ed1cea246
MD5 (/Users/stefanodimarino/Library/Application Support/OpenEmu/BIOS/SCPH5502.bin) = 32736f17079d0b2b7024407c39bd3050

Spyro - Year of the Dragon (v1.1).bin
Spyro - Year of the Dragon (v1.1).cue

FILE "Spyro - Year of the Dragon (v1.1).bin" BINARY
  TRACK 01 MODE2/2352
    INDEX 01 00:00:00

https://gist.github.com/cresi/68134f1b520153df08d6

**Comment by clobber** [comment]
Okay, thanks for all of the info. The 0.9.38.2 version uploaded had some API from an unreleased version of OE. I've fixed the build and re-uploaded it here: https://github.com/OpenEmu/Mednafen-Core/releases/download/v0.9.38.2/Mednafen_0.9.38.2.zip

That one should now run for you. Please test thoroughly and let me know if the issue with the slowdown persists. As for filters, just use Pixellate.

**Comment by clobber** [comment]
Any change?

**Comment by cresi** [comment]
Ok, sorry but this weekend was full of stuff I couldn't try it out. Anyway I installed the core you gave me but nothing changed the problem still exists. (tried Spyro and FF VII)

**Comment by clobber** [comment]
Which filter are you using?

**Comment by cresi** [comment]
Oh yes, forgot to switch to Pixellate. Tried to run the game, works super fine. Thank you for your help, i'll try to test everything to the full and let you know if some other issue happens.

**Comment by clobber** [comment]
And which filter were you using before? That could have been the issue the entire time.

**Comment by cresi** [comment]
Nearest Neighbor

**Comment by RoastyMcToasty** [comment]
I am using that too but have never had any issues..

**Comment by clobber** [comment]
@RoastyMcToasty Thanks but that isn't particularly helpful. Also, please don't bump other tickets. I'll report back when there is something to report.

@cresi That's odd. Nearest Neighbor should be the least of your problems in terms of filters.

**Comment by clobber** [comment]
Closing since appears to have been a problem with an older build of the core. Report back if it happens again.

**Comment by cresi** [comment]
Ok i have been playing for a couple of days, it worked fine till now when i ran openemu with teamspeak opened and headphones plugged in. The issue reappeared...could it be teamspeak the problem here, because when it appeared the first time I was running teamspeak?

**Comment by clobber** [comment]
Not conclusive unless you can replicate the problem every time with teamspeak.

**Comment by cresi** [comment]
Ok, the problem started all over again. I tried switching between filters, i tried to reinstall openemu with the new core, but nothing seems to fix the issue.

**Comment by clobber** [comment]
Problem starts how exactly? When running other software like teamspeak? You shouldn't be taking all those steps because that makes it more difficult to track down the problem. 'Reinstalling' won't do anything. It's not like the application can break. For filters, you should only be using Pixellate or maybe Nearest Neighbor if that causes no problem.

Try again using this test build of OE https://transfer.sh/fLiFc/openemu-1.0.4-experimental-no-clientstorage.zip

and with Mednafen 0.9.38.2 https://github.com/OpenEmu/Mednafen-Core/releases/download/v0.9.38.2/Mednafen_0.9.38.2.zip

Then report back if there is any change.

**Comment by cresi** [comment]
Ok yesterday i tried to use the test build, wouldn't work. Today i gave it another shot, now works. I don't know why it does this...

**Comment by clobber** [comment]
Can you elaborate? What does wouldn't work mean? The app wouldn't launch or something else? Is there a crash log? What works now? PSX? ...

**Comment by cresi** [comment]
Ok, I'll clear it out. Yesterday when i ran a psx game, the issue i opened this thread for presented it again. I installed the test build you linked but the issue didn't stop. Today i gave it another shot and the issue stopped. Now, while i'm writing this post, i have OE running a game, steam, teamspeak and safari opened but no stutters. The problem is fixed now but i don't know if it can reappear like yesterday. Hope it's clearer now.

**Comment by clobber** [comment]
If the problem didn't first go away with the new build then it doesn't sound fixed. If it happens again let me know and there will be one last build we can try with.

**Comment by cresi** [comment]
Right now it seems stable, today for example i tried to play a game but it stuttered, then i waited a couple of hours, replayed it and it worked fine. Maybe it's better to wait further releases :D

**Comment by clobber** [comment]
Okay, here's a new version that has been compiled a little differently: https://transfer.sh/13oa1g/mednafen-0.9.38.3.zip

Same drill as last time: unzip and place in your Cores folder, report back any problems.

**Comment by monsieurhub** [comment]
I have the exact same problem when using any PlayStation game. In fact, the only way for the games not to stutter and slow down is to play them in a window and not full screen. If played full screen, the game becomes unplayable because of the stuttering and sound glitches. I tried using Mednafen 0.9.38.2 and it did not change anything. 

My computer is a Macbook Air with these specs :
1.8 ghz Core i7
4 gig RAM
Intel HD Graphics 3000 384 Mo

I tried using many roms. The one I use to test is Metal Gear Solid since the opening cinematics is very intensive.
